### PR TITLE
some unit tests for ConfigManager

### PR DESCRIPTION
**Pull request by alikins**
test the case for:
https://github.com/ansible/ansible/pull/30651
https://github.com/ansible/ansible/issues/30633

##### ISSUE TYPE

 
 - Tests Pull Request

##### COMPONENT NAME

test/units/config
##### ANSIBLE VERSION

```
ansible 2.5.0 (config_unit_tests 6a72116a0b) last updated 2017/09/20 17:29:15 (GMT -400)
  config file = /home/adrian/src/ansible/ansible.cfg
  configured module search path = [u'/home/adrian/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adrian/src/ansible/lib/ansible
  executable location = /home/adrian/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]


```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/units/config/test_manager.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        ini_config = configparser.ConfigParser()
+        for item in contents:
+            ini_config.add_section(item['section'])
+            ini_config.set('defaults', item['name'], item['value'])
+
+        ini_config.write(self.tmp_file)
+        self.tmp_file.close()
+        return self.tmp_file
+
+    def test_hostfile(self):
```
Comment (bcoca) [inline test/units/config/test_manager.py:30]: the tests should not really be for hostfile, as we are going to remove it.

a mock setting with 2 entries would test the functionality, not the specific setting

File: test/units/config/test_manager.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        cm = manager.ConfigManager(conf_file=_cfg_file.name)
+        hostfile = cm.data.get_setting('DEFAULT_HOST_LIST')
+        self.assertEqual(hostfile.value, ['/dev/null/a.yml'])
+
+    def test_inventory_ignore_patterns(self):
+        _cfg_file = self._config_file(contents=[{'section': 'defaults',
+                                                 'name': 'inventory_ignore_patterns',
+                                                 'value': '/dev/null/foo,/dev/null/bar'}])
+        cm = manager.ConfigManager(conf_file=_cfg_file.name)
+        inventory_patterns = cm.data.get_setting('INVENTORY_IGNORE_PATTERNS')
```
Comment (bcoca) [inline test/units/config/test_manager.py:43]: same here, we really dont want to test the particular setting, just that both the default (dynamic template) and the nii entry are correctly parsed.

**Comment by sivel** [comment]
I think you'll need an `__init__.py` in that dir too.

**Comment by mattclay** [comment]
CI failure in Python 3.x unit tests:

https://app.shippable.com/github/ansible/ansible/runs/38144/4/tests

**Comment by alikins** [comment]
@sivel wouldn't hurt to add it, but ansible-test/pytest doesn't seem to need the __init__ for test discovery

**Comment by alikins** [comment]
Also, these tests aren't isolated from local ansible.cfg settings or environment variables.  But thats not new, none of the unit tests are (aside from a few that patch ansible.constants or the local 'C').  

ConfigManager() itself pulls from env unless it  _loop_entries or os.environ are mocked, but thats the kind of stuff having any unit tests for config are meant to find.

**Comment by mattclay** [comment]
@alikins The tests should be isolated from environment variables, where are you seeing otherwise?

The tests are also supposed to be isolated from config, but a change to how config is handled broke that. I'll work on a fix.

**Comment by alikins** [comment]
shrug

**Comment by mattclay** [comment]
The test config isolation was fixed in https://github.com/ansible/ansible/pull/31325

**Pull request closed without merging**